### DISCUSSION TITLE
Widget Self Referencing

### DISCUSSION THREAD

**Discussion by Acetolyne** [fyne-io/fyne#4392] [Q&A] [answered]
Hi team, 

I have tried to find the answer elsewhere but couldn't 
I was wondering if there is a way to self reference a widget. I am trying to make a form where onCancel would hide itself.
The form is hidden by default until a button is pressed but if the form is canceled I would like to hide the form again.

```
deviceForm := &widget.Form{
		Items: []*widget.FormItem{
			{Text: "Name", Widget: name},
			{Text: "URL", Widget: url},
		},
		OnSubmit: func() {
			log.Println("device:", name.Text)
			log.Println("url:", url.Text)
		},
		OnCancel: func() {
			HIDE deviceForm <<<<<<<<<<<<<<<<-----**How can I accomplish this? It cannot reference itself**
			//fmt.Println("Hide the form")
		},
		SubmitText: "Add Device",
		CancelText: "Cancel",

	}
	deviceForm.Hide()
```

Thanks for any guidance

**Comment by Jacalz** [comment]
This is the wrong repository. I suppose you meant to open in fyne-io/fyne.
FYI: We also want questions to use discussions, not issues.

**Comment by Jacalz** [comment, marked as answer]
To answer your question:

You just set the OnCancel method later:
```go
deviceForm := &widget.Form{
		Items: []*widget.FormItem{
			{Text: "Name", Widget: name},
			{Text: "URL", Widget: url},
		},
		OnSubmit: func() {
			log.Println("device:", name.Text)
			log.Println("url:", url.Text)
		},
		SubmitText: "Add Device",
		CancelText: "Cancel",

}
deviceForm.OnCancel = func() {
	deviceForm.Hide()
	fmt.Println("Hide the form")
}
deviceForm.Hide()

```

**Reply by Acetolyne** [reply]
I did mean to use the fyne.io repository sorry for that but greatly appreciate the help.

**Reply by Jacalz** [reply]
No problem. GitHub provides tools to move issues nowadays. I'm glad it solved your question :)
